### ISSUE TITLE
Weights don't have any effect.

### ISSUE THREAD

**Issue by stur86** [Kcnarf/d3-weighted-voronoi#2]
I did a basic experiment with the library on CodePen but weights don't seem to affect the result in the slightest. Any idea why? Thanks!

https://codepen.io/stur86/pen/BdXGWO

**Comment by Kcnarf** [comment]
TL;DR; : adjust/scale weights so that they have significant effect. For your example, use 10000 and 2000 (instead of 1 and 0.2).

Below are some deeper explanations (of the same thing) with different approaches.
* considering 2 sites with each a weight of 10; if the sites are close (eg. 2px), setting 1 weight to 5 will have a huge impact; but if the sites are far (eg. 100px), doing the same thing (1 weight to 5) will have near-zero impact
* The weight of a site has less and less impact as the distance grows;
* a site with the default weight (1), has a default influence area of 1 px; to increase its influence area to 2 px, its weight must be 4 (depends on square root); And then 10 px influence area requires a weight of 100; 100px influence area <= weight of 10000. You can experiment this in [Voronoï playground : interactive weighted Voronoï study](http://blockbuilder.org/Kcnarf/dacd1d9d2f0e69cf93c68ecf32f7896d) by increasing the weight of 1 site, 1 unit at a time. You will see that the weight must be increased in a square manner to increase the influence area of a site. For your example, your viz is ~200 px wide, so weights must be scaled in order to corresponds to this situation.

**Comment by stur86** [comment]
Oh, thanks, I got it. But wouldn't it make more sense to have the weights be normalised somehow? If I hear "weights" I think all that matters is the *ratio* between them, not their absolute values.

**Comment by Kcnarf** [comment]
I understand.
The adjustment depends on final viz size and the number of sites:
 * considering a set of weighted sites, adjustment depends on the size of the final viz: larger size requires higher weights to 'fill the gap'
* conversely , for a predefined size of the final viz, higher number of sites requires lower weights (because there is less gap between sites); if weights are too high, the risk is to have overweighted sites with no influence area.

As a rule a thumb, I found that the adjustment *vizArea/numberOfSites* does the trick:
* for rectangle, adjustment is *widht\*height/numberOfSites*
* for triangle, adjustment is *widht\*height/2/numberOfSites*
* for circle, adjustment is *PI\*r\*r/numberOfSites*
* for other types of convex clipping polygon, one could use [d3.polygonArea(*polygon*)](https://github.com/d3/d3-polygon#polygonArea)

But I'm not confortable with this rule. Hence, for the moment and before further investigations/understandings, I prefer to let this decision to the end-user. I guess I can add this discussion to the README.

**Comment by Kcnarf** [comment]
Not an issue, just a how-to
